shared/src/business/useCases/processStreamRecordsInteractor.js

```diff
@@ -248,10 +256,72 @@ const processWorkItemEntries = async ({
   });
 
   if (failedRecords.length > 0) {
-    applicationContext.logger.error(
-      'the records that failed to index',
+    applicationContext.logger.error('the records that failed to index', {
       failedRecords,
-    );
+    });
+    throw new Error('failed to index records');
+  }
+};
+
+const processMessageEntries = async ({
+  applicationContext,
+  messageRecords,
+  utils,
+}) => {
+  if (!messageRecords.length) return;
+
+  applicationContext.logger.debug(
+    `going to index ${messageRecords.length} message records`,
+  );
+
+  const indexMessageEntry = async messageRecord => {
+    const messageNewImage = messageRecord.dynamodb.NewImage;
+
+    // go get the latest message if we're indexing a message with isRepliedTo set to false - it might
+    // have been updated in dynamo since this record was created to be processed
+    if (!messageNewImage.isRepliedTo.BOOL) {
+      const latestMessageData = await utils.getMessage({
+        applicationContext,
+        docketNumber: messageNewImage.docketNumber.S,
+        messageId: messageNewImage.messageId.S,
+      });
+
+      const marshalledMessage = AWS.DynamoDB.Converter.marshall(
+        latestMessageData,
+      );
+
+      return {
+        dynamodb: {
+          Keys: {
+            pk: {
+              S: messageNewImage.pk.S,
+            },
+            sk: {
+              S: messageNewImage.sk.S,
+            },
+          },
+          NewImage: marshalledMessage,
+        },
+        eventName: 'MODIFY',
+      };
```

Comment:
Just a gut check here. Do we even need to do _anything_ if the fetched DDB record has `isRepliedTo === true`. It would seem we would just wait until that event made it through Kinesis (or maybe it already has somehow?)

Reply:
I can't think of any reason why we have to. Would you like for me to make that change?

Reply:
Yes, please. However, in the future should we still have messages sticking around, we may opt for this method that covers that base thoroughly.